mmdet3d/apis/inference.py

```diff
@@ -106,6 +109,81 @@ def inference_detector(model, pcd):
     return result, data
 
 
+def inference_multi_modality_detector(model, pcd, image, ann_file):
+    """Inference point cloud with the multimodality detector.
+
+    Args:
+        model (nn.Module): The loaded detector.
+        pcd (str): Point cloud files.
+        image (str): Image files.
+        ann_file (str): Annotation files.
+
+    Returns:
+        tuple: Predicted results and data from pipeline.
+    """
+    cfg = model.cfg
+    device = next(model.parameters()).device  # model device
+    # build the data pipeline
+    test_pipeline = deepcopy(cfg.data.test.pipeline)
+    test_pipeline = Compose(test_pipeline)
+    box_type_3d, box_mode_3d = get_box_type(cfg.data.test.box_type_3d)
+    # get data info containing calib
+    data_infos = mmcv.load(ann_file)
+    image_idx = int(image[-10:-4])
+    for x in data_infos:
+        if int(x['image']['image_idx']) != image_idx:
+            continue
```

Comment:
A little bit hard code, including the line 132? Can we compare the filename or something?

Reply:
Use regular expression to match the img_idx here.